### PR TITLE
chore: Run CI on PHP 8.3 and 8.4, fix PHP 8.4 support, pin actions versions

### PR DESCRIPTION
**Pull request by GrahamCampbell**
It looks like the generated code has some compatibility issues with PHP 8.4. For example `_DictionaryGetResponse $response = null` is not allowed without a `?` in front, to explicitly indicate the type is nullable. Would it be permissible to run the codebase through php-cs-fixer to fix compatibility issues post-code-gen?

### CODE REVIEW

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -12,9 +12,9 @@ permissions:
 
 jobs:
   dependency-review:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
```
Comment (cprice404) [inline .github/workflows/dependency-review.yml:15]: we've had problems in the past with ubuntu 22.04 and some networking issues between github/azure and momento. We can pin these to 24.04 if you like.

File: .github/workflows/on-pull-request.yaml [hunk 2]
```diff
@@ -12,14 +12,14 @@ jobs:
       max-parallel: 1
       fail-fast: true
       matrix:
-        php-version: ["7.4", "8.0", "8.1", "8.2"]
-    runs-on: ubuntu-latest
+        php-version: ['7.4', '8.0', '8.1', '8.2', '8.3', '8.4']
```
Comment (cprice404) [inline .github/workflows/on-pull-request.yaml:15]: definitely in favor of this, thank you.

**Review by cprice404 — COMMENTED** [review]
thanks! and yes we are happy to run the linter/fixer if it won't break compat with older versions! i'd be happy to add them to our CI jobs and build scripts too to enforce it.

**Comment by GrahamCampbell** [comment]
I've added in the linting to the CI. It can be run locally without `--dry-run` to actually apply the fixes.

File: .github/workflows/dependency-review.yml (hunk 1 above)
Comment (GrahamCampbell) [reply .github/workflows/dependency-review.yml]: Updated

**Comment by GrahamCampbell** [comment]
Replaced by #211.

**Pull request closed without merging**